### PR TITLE
Build on Pull-request and deploy on push

### PR DESCRIPTION
**Pull request by artemkovalyov**
## Context

PoC for documentaiton pipelines

### CODE REVIEW

**Comment by artemkovalyov** [comment]
@fwilhe , it's my first try. Let's develop it into smth feasible.

File: .github/workflows/build-documentation.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: Build Documentation
+
+on:
+  pull_request:
+    branches-ignore: [master, gh-pages]
+    branches: [documentation]
```
Comment (marikaner) [inline .github/workflows/build-documentation.yml:6]: Does it make sense to have both here? I assume you can delete line 5.

File: .github/workflows/deploy-documentation.yml [hunk 2]
```diff
@@ -0,0 +1,25 @@
+name: Build Documentation
```
Comment (marikaner) [inline .github/workflows/deploy-documentation.yml:1]: The name is duplicate here.

File: .github/workflows/deploy-documentation.yml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+
+on:
+  push:
+    branches-ignore: [master, gh-pages]
+    branches: [documentation]
```
Comment (marikaner) [inline .github/workflows/deploy-documentation.yml:6]: Same as above. You can remove `branches-ignore`.

File: .github/workflows/deploy-documentation.yml [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+jobs:
+  deploy:
+    runs-on: ubuntu-latest
+    timeout-minutes: 15
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          ref: documentation
```
Comment (marikaner) [inline .github/workflows/deploy-documentation.yml:15]: I think this is the default and can be left out.

**Comment by artemkovalyov** [comment]
I'll rework this.

**Pull request closed without merging**